.github/workflows/release.yml.off

```diff
@@ -1,21 +1,66 @@
 # (c) Copyright 2019-2023 OLX
 
-name: build
+name: release
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}/dali
 
 on:
   push:
     branches:
       - master
-    paths:
-      - Cargo.toml
+    tags:
+      - 'v*'
+
 jobs:
-  publish_docker_image:
+  release:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - name: Docker login
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
-      - name: Adds execution permissions to the script which extracts the current version
-        run: chmod +x scripts/get-current-version.sh
-      - name: Build and Publish to GPR
-        run: make docker-publish
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Semantic Release
+        id: semantic
+        uses: cycjimmy/semantic-release-action@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_USER: ${{ github.repository_owner }}
+        with:
+          dry_run: false
+          extra_plugins: |
+            @semantic-release/changelog@v6.0.3
+            @semantic-release/git@10.0.1
+            @semantic-release/github@9.2.0
+            @semantic-release/commit-analyzer@11.1.0
+            @semantic-release/release-notes-generator@12.1.0
+            semantic-release-major-tag@0.3.2
+
+      - name: Log in to the Container registry
+        if: steps.semantic.outputs.new_release_published == 'true'
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        if: steps.semantic.outputs.new_release_published == 'true'
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: olxgroup-oss/dali/dali
```

Comment:
Shouldn't we use here the `IMAGE_NAME` env variable defined at the top of the file?

Reply:
I'm not entirely sure what this variable will evaluate to. I've made a change for the name that will be pushed shortly